qcodes/instrument_drivers/Galil/dmc_41x3.py

```diff
@@ -0,0 +1,1033 @@
+        self._move()
+
+    def move_towards_left_bottom_position(self) -> None:
+
+        self._pick_up()
+
+        motion_vec = -1 * self._a
+        self._setup_motion(rel_vec=motion_vec, d=self.norm_a, speed=self._speed)
+        self._move()
+        self._put_down()
+
+        self._current_row = 1
+        self._current_pad = 1
+
+    def move_to_next_row(self) -> None:
+
+        if self._current_row is None or self._current_pad is None:
+            raise RuntimeError("Current position unknown.")
+
+        if self._current_row == self.rows:
+            raise RuntimeError("Cannot move further")
+
+        self._pick_up()
+
+        self._setup_motion(rel_vec=self._b,
+                           d=self.inter_row_distance,
+                           speed=self._speed)
+        self._move()
+
+        self._put_down()
+
+        self._current_row = self._current_row + 1
+
+    def move_to_begin_row_pad_from_end_row_last_pad(self) -> None:
+
+        if self._current_row is None or self._current_pad is None:
+            raise RuntimeError("Current position unknown.")
+
+        if self._current_pad == self.pads:
+            raise RuntimeError("Cannot move further")
+
+        motion_vec = -1 * self._b * self.norm_b + self._c * \
+                     self.inter_pad_distance
+        norm = np.linalg.norm(motion_vec)
+        motion_vec_cap = motion_vec / norm
+
+        self._pick_up()
+
+        self._setup_motion(rel_vec=motion_vec_cap, d=norm, speed=self._speed)
+        self._move()
+
+        self._put_down()
+
+        self._current_row = 1
+        self._current_pad = self._current_pad + 1
+
+    def move_to_row(self, num: int) -> None:
+
+        if num < 1 or num > self.rows:
+            raise RuntimeError(
+                f"Row num: {num} is out of range. Row numbers start from 1 "
+                f"and max number of rows is {self.rows}."
+            )
+
+        sign = 1
+        assert self._current_row is not None
+        d = (num - self._current_row) * self.inter_row_distance
+
+        if d == 0:
+            raise RuntimeError(
+                f"You are at the row where you want to be. Current row number "
+                f"is {self._current_row}."
+            )
+        if d < 0:
+            sign = -1
+            d = abs(d)
+
+        self._pick_up()
+
+        self._setup_motion(rel_vec=sign * self._b, d=d, speed=self._speed)
+        self._move()
+
+        self._put_down()
+
+        self._current_row = num
+
+    def move_to_pad(self, num: int) -> None:
+
+        if num < 1 or num > self.pads:
+            raise RuntimeError(
+                f"Pad num: {num} is out of range. Pad number start from 1 "
+                f"and max number is {self.pads}."
+            )
+
+        sign = 1
+        assert self._current_pad is not None
+        d = (num - self._current_pad) * self.inter_pad_distance
+
+        if d == 0:
+            raise RuntimeError(
```

Comment:
it would be good to ask users whether they like a runtime exception here or just let the method return if the current pad is already the desired one. My hypothesis is that an exception is not desirable, but i might be wrong :)

Reply:
ok, will check this :)

Reply:
I checked this, and preference is for letting the users know that are in fact they are on the row that they want to move to. So, current implementation can stay.

Reply:
perfect :)